src/r_main.c

```diff
@@ -621,15 +605,43 @@ angle_t R_InterpolateAngle(angle_t oangle, angle_t nangle, fixed_t scale)
     }
 }
 
+void R_SetupMouselook(void)
+{
+  static int old_viewpitch, old_viewheight;
+  fixed_t dy;
+  int i;
+
+  if (viewpitch != old_viewpitch || viewheight != old_viewheight)
+  {
+    old_viewpitch = viewpitch;
+    old_viewheight = viewheight;
+  }
+  else
+  {
+    return;
+  }
+
+  viewpitch = BETWEEN(MIN_VIEWPITCH, MAX_VIEWPITCH, viewpitch);
+
+  dy = FixedMul(projection, finetangent[(ANG90 - viewpitch) >> ANGLETOFINESHIFT]);
+  centery = viewheight / 2 - (dy >> FRACBITS);
+  centeryfrac = centery << FRACBITS;
+
+  for (i = 0; i < viewheight; i++)
```

Comment:
So, this whole array needs to be recalculated each time the pitch changes?

Reply:
Yes, but it's a much smaller array than `yslopes[LOOKDIRSMAX][viewheight]`. From my tests, it seems to be fast.

Reply:
It still has to iterate over the entire vertical resolution, i.e. 900p on my display.

Reply:
On my old laptop it runs in ~12 microseconds (us). I haven't noticed any change in the average frametime. The old method is faster, but I think the higher accuracy is worth it. Also, we don't need to pre-calculate large `yslopes` table for it.

Reply:
To make sure we don't mix things up: In Crispy, the `yslopes[][]` array contains precalculated `yslope[]` arrays for each possible `pitch` value. It only needs to be recalculated if the resolution changes! I introduced it, because I did not want to recalculate `yslope[]` each time the pitch changes (as is done e.g. in Vanilla Heretic, but this only has keyboard look and 200p resolution). Instead, `yslope` has become a mere pointer into the `yslopes` array.

I'd still prefer this method, but admit that it doesn't scale well with dynamic resolution switching.

Reply:
> It only needs to be recalculated if the resolution changes! 

And also when the FOV is changed. Anyway, the problem with our old approach is that we need the set of possible values of `pitch` to be very large, so a pre-computed array is impractical.

> I introduced it, because I did not want to recalculate `yslope[]` each time the pitch changes (as is done e.g. in Vanilla Heretic, but this only has keyboard look and 200p resolution).

I thought `yslopes[][]` was from Heretic. So it was fast enough even in 1994.

> I'd still prefer this method, but admit that it doesn't scale well with dynamic resolution switching.

Yeah, that's the main reason for me. Also, Ceski's "unlagged" mouse code requires more precision.